ObjectMapper/Core/Mapper.swift

```diff
@@ -102,6 +102,13 @@ public final class Mapper<N: BaseMappable> {
 			}
 		}
 
+		if let klass = N.self as? ImmutableMappable.Type {
+			if let maybeObject = try? klass.init(map: map) as? N, var object = maybeObject {
```

Comment:
Hmmm. @nmccann @tristanhimmelman let's think about this:

1. `mapArray()` should throw error when there's invalid JSON model

    ```swift
    let array = [validJSON_1, invalidJSON_2, validJSON_3]
    let models = try mapArray(JSONArray: array) // should fail?
    ```

    With this approach, `mapArray()` can only handle `ImmutableMappable`s.

2. `mapArray()` should ignore invalid JSON model

    ```swift
    let array = [validJSON_1, invalidJSON_2, validJSON_3]
    let models = mapArray(JSONArray: array) // should return [model_1, model_3]?
    ```

    With this approach, `mapArray()` can handle any `BaseMappable`s.